cmd/frontend/graphqlbackend/search.go

```
@@ -883,19 +883,32 @@ func resolveRepositories(ctx context.Context, op resolveRepoOp) (repoRevisions,
-			} else if isDefaultBranch := rev.RevSpec == ""; !isDefaultBranch { // skip default branch resolution to save time
-				// Validate the revspec.
-
-				// Do not trigger a repo-updater lookup (e.g.,
-				// backend.{GitRepo,Repos.ResolveRev}) because that would slow this operation
-				// down by a lot (if we're looping over many repos). This means that it'll fail if a
-				// repo is not on gitserver.
-				//
-				// TODO(sqs): make this NOT send gitserver this revspec in EnsureRevision, to avoid
-				// searches like "repo:@foobar" (where foobar is an invalid revspec on most repos)
-				// taking a long time because they all ask gitserver to try to fetch from the remote
-				// repo.
-				if _, err := git.ResolveRevision(ctx, repoRev.GitserverRepo(), nil, rev.RevSpec, git.ResolveRevisionOptions{NoEnsureRevision: true}); gitserver.IsRevisionNotFound(err) || err == context.DeadlineExceeded {
+				repoRev.Revs = append(repoRev.Revs, rev)
+				continue
+			}
+			if rev.RevSpec == "" { // skip default branch resolution to save time
+				repoRev.Revs = append(repoRev.Revs, rev)
+				continue
+			}
+
+			// Validate the revspec.
+			// Do not trigger a repo-updater lookup (e.g.,
+			// backend.{GitRepo,Repos.ResolveRev}) because that would slow this operation
+			// down by a lot (if we're looping over many repos). This means that it'll fail if a
+			// repo is not on gitserver.
+			//
+			// TODO(sqs): make this NOT send gitserver this revspec in EnsureRevision, to avoid
+			// searches like "repo:@foobar" (where foobar is an invalid revspec on most repos)
+			// taking a long time because they all ask gitserver to try to fetch from the remote
+			// repo.
+			if _, err := git.ResolveRevision(ctx, repoRev.GitserverRepo(), nil, rev.RevSpec, git.ResolveRevisionOptions{NoEnsureRevision: true}); err != nil {
+				if errors.Is(err, context.DeadlineExceeded) {
+					return nil, nil, false, nil, context.DeadlineExceeded
+				}
+				if errors.As(err, &git.BadCommitError{}) {
```

previously we ignored this error which led to search results that returned a 404 when clicked upon.